api/PayrollProcessor.Data.Persistence/Features/Employees/EmployeePayrollUpdateCommandHandler.cs

```diff
@@ -24,25 +25,24 @@ public EmployeePayrollUpdateCommandHandler(CosmosClient client, IQueueClientFact
             queueClient = clientFactory.Create(AppResources.Queue.EmployeePayrollUpdates);
         }
 
-        public TryOptionAsync<EmployeePayroll> Execute(EmployeePayrollUpdateCommand command, CancellationToken token) =>
-            async () =>
-            {
-                var record = EmployeePayrollRecord.Map.Merge(command); ;
-
-                var updateResponse = await client
+        public TryAsync<EmployeePayroll> Execute(EmployeePayrollUpdateCommand command, CancellationToken token) =>
+            command.Apply(EmployeePayrollRecord.Map.Merge)
```

Comment:
Not sure which style I prefer. This could also be written as

```csharp
EmployeePayrollRecord.Map.Merge(command)
  .Apply(record => client...)
```